src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets

```diff
@@ -63,8 +62,27 @@ Copyright (c) .NET Foundation. All rights reserved.
     </BuildOutputInPackage>
   </ItemDefinitionGroup>
 
-  <Target Name="_GetOutputItemsFromPack"
-          Returns="@(_OutputPackItems)">
+  <PropertyGroup>
+    <PackageOutputPath Condition="'$(PackageOutputPath)' == ''">$(BaseOutputPath)$(Configuration)\</PackageOutputPath>
+    <NuspecOutputPath Condition="'$(NuspecOutputPath)' == ''">$(BaseIntermediateOutputPath)$(Configuration)\</NuspecOutputPath>
+  </PropertyGroup>
+
+  <!--
+    ============================================================
+    _GetAbsoluteOutputPathsForPack
+    Gets the absolute output paths for Pack.
+    ============================================================
+  -->
+  <Target Name="_GetAbsoluteOutputPathsForPack">
+    <PropertyGroup>
+      <RestoreOutputPath Condition="'$(RestoreOutputPath)' == ''">$(MSBuildProjectExtensionsPath)</RestoreOutputPath>
+      <PackageOutputPath Condition="'$(PackageOutputPath)' == ''">$(BaseOutputPath)$(Configuration)\</PackageOutputPath>
```

Comment:
Do we need to apply the default in both places? Or is 1 enoguh?

Reply:
within `_GetAbsoluteOutputPathsForPack` is enough but I'm afraid that we might regress something that haven't been properly tested, that's why I added them globally too! (as `NuSpecOutputPath` was already there).

Also one benefit (side effect) is that the defaults will be available after MSBuild's property pass.

Can anyone clarify this?

Reply:
@rainersigwald 
Any gotchas with properties declared inside a target vs globally? 

Specifically any concerns with moving the output defaults from inside a target where they are used to global. 
The expectation is what the output defaults to is something that doesn't change. 

I *think* adding them globally is safe.

Reply:
Properties defined inside targets don't get set until the target runs, which can present ordering issues. In this case, since they're unconditionally defaulted outside a target, it'd be pretty funky (though certainly possible) for them to be unset when the target started. A project would have to explicitly unset them either in a later .targets or in a target.

The biggest problem with leaving them only as evaluation-time targets is that they can be modified by user code. But that's also a configurability advantage--and this PR only sets defaults, not a problem, hopefully.

I lean toward "only outside the target".

Also, a big nit to pick with y'all :): [global properties](https://docs.microsoft.com/visualstudio/msbuild/msbuild-properties?view=vs-2019#global-properties) are a thing, but not something that's on the table here. You scared me at first

Reply:
> > I lean toward "only outside the target".

Thanks for the explanation. @nirmal4g let's do outside of the target only.

> Also, a big nit to pick with y'all :): global properties are a thing, but not something that's on the table here. You scared me at first

😆 I appreciate the nit pick!
Life is simpler when we use the vernacular 😄